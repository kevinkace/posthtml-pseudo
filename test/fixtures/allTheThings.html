<!doctype html>
<html>
<head>
    <title>page title</title>
</head>
<body>
    <div>
        <h1>header</h1>
        <p>para</p>
    </div>

    <form>
        <input type="text" name="name" class="">
        <input type="text" class="" name="name" />
        <input type="text" name="name" enabled>
        <input class="name" type="text" name="name" disabled>

        <input type="submit" name="submit">

        <button>button</button>
    </form>

    <div class="article">
        <h1>Proin ex erat, semper vel tincidunt et</h1>
        <p>
            Lorem ipsum dolor sit amet, consectetur <a href="#jump">adipiscing elit</a>. Aenean et varius nunc, accumsan aliquam ligula. Donec tempus nibh sit amet ante porta, ut posuere lorem commodo.
        </p>
        <p>
            Suspendisse id eros mollis, sollicitudin nisl ac, tristique risus. Quisque placerat, nisl quis sodales imperdiet, justo tellus vehicula lorem, eget dignissim eros mi ac ligula. Suspendisse ultrices magna at erat consequat egestas. Nunc congue ex quis nulla rutrum varius. Sed lobortis posuere ipsum, porttitor laoreet risus fringilla id. Donec non enim consequat, finibus diam quis, pulvinar sem.
        </p>

        <div>
            <ul>
                <li>Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Donec vitae eros suscipit, ultrices metus vitae, euismod nulla.</li>
                <li>Nullam eu tempor massa.</li>
                <li>Vivamus egestas nisi mauris, vitae varius mi vehicula vitae.</li>
                <li>Mauris volutpat ullamcorper justo, quis tempor turpis dapibus iaculis.</li>
            </ul>
        </div>

        <p>
            Vivamus eget tempor nulla, sit amet interdum purus. Ut ullamcorper, orci in feugiat cursus, augue neque rutrum velit, at rhoncus ipsum leo ac purus. Nulla felis massa, pharetra sit amet neque sed, sollicitudin finibus leo. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae;
        </p>
    </div>
    <nav>
        <a href="http://example.com">example</a>
        <a href="http://example.com">example 2</a>
        <a href="http://example.com">not kexample</a>
    </nav>
</body>
</html>
