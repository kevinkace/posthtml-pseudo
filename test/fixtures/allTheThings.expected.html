<!doctype html>
<html>
<head>
    <title>page title</title>
</head>
<body>
    <div>
        <h1 class="firstchild-bookend">header</h1>
        <p class="lastchild-bookend">para</p>
    </div>

    <form>
        <input type="text" name="name" class=":enabled :optional firstchild-bookend hidden">
        <input type="text" class=":enabled :optional hidden" name="name">
        <input type="text" name="name" enabled="" class=":enabled :optional hidden">
        <input class="name :disabled :optional hidden" type="text" name="name" disabled="">

        <input type="submit" name="submit" class=":default :enabled :optional hidden">

        <button class=":enabled :optional lastchild-bookend">button</button>
    </form>

    <div class="article">
        <h1 class="firstchild-bookend">Proin ex erat, semper vel tincidunt et</h1>
        <p>
            Lorem ipsum dolor sit amet, consectetur <a href="#jump" class=":enabled firstchild-bookend lastchild-bookend">adipiscing elit</a>. Aenean et varius nunc, accumsan aliquam ligula. Donec tempus nibh sit amet ante porta, ut posuere lorem commodo.
        </p>
        <p>
            Suspendisse id eros mollis, sollicitudin nisl ac, tristique risus. Quisque placerat, nisl quis sodales imperdiet, justo tellus vehicula lorem, eget dignissim eros mi ac ligula. Suspendisse ultrices magna at erat consequat egestas. Nunc congue ex quis nulla rutrum varius. Sed lobortis posuere ipsum, porttitor laoreet risus fringilla id. Donec non enim consequat, finibus diam quis, pulvinar sem.
        </p>

        <div>
            <ul class="firstchild-bookend lastchild-bookend">
                <li class="firstchild-bookend">Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Donec vitae eros suscipit, ultrices metus vitae, euismod nulla.</li>
                <li>Nullam eu tempor massa.</li>
                <li>Vivamus egestas nisi mauris, vitae varius mi vehicula vitae.</li>
                <li class="lastchild-bookend">Mauris volutpat ullamcorper justo, quis tempor turpis dapibus iaculis.</li>
            </ul>
        </div>

        <p class="lastchild-bookend">
            Vivamus eget tempor nulla, sit amet interdum purus. Ut ullamcorper, orci in feugiat cursus, augue neque rutrum velit, at rhoncus ipsum leo ac purus. Nulla felis massa, pharetra sit amet neque sed, sollicitudin finibus leo. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae;
        </p>
    </div>
    <nav class="lastchild-bookend">
        <a href="http://example.com" class=":enabled firstchild-bookend">example</a>
        <a href="http://example.com" class=":enabled">example 2</a>
        <a href="http://example.com" class=":enabled lastchild-bookend">not kexample</a>
    </nav>
</body>
</html>
